packages/client/mutations/UpsertTeamPromptResponseMutation.ts

```diff
@@ -0,0 +1,96 @@
+import graphql from 'babel-plugin-relay/macro'
+import {commitMutation} from 'react-relay'
+import {StandardMutation} from '../types/relayMutations'
+import clientTempId from '../utils/relay/clientTempId'
+import createProxyRecord from '../utils/relay/createProxyRecord'
+import {UpsertTeamPromptResponseMutation as TUpsertTeamPromptResponseMutation} from '../__generated__/UpsertTeamPromptResponseMutation.graphql'
+
+graphql`
+  fragment UpsertTeamPromptResponseMutation_meeting on UpsertTeamPromptResponseSuccess {
+    teamPromptResponse {
+      id
+      content
+      plaintextContent
+    }
+    meeting {
+      ... on TeamPromptMeeting {
+        id
+        createdAt
+        phases {
+          ... on TeamPromptResponsesPhase {
+            stages {
+              ... on TeamPromptResponseStage {
+                ...TeamPromptResponseCard_stage
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`
+
+const mutation = graphql`
+  mutation UpsertTeamPromptResponseMutation(
+    $teamPromptResponseId: ID
+    $meetingId: ID!
+    $content: String!
+  ) {
+    upsertTeamPromptResponse(
+      teamPromptResponseId: $teamPromptResponseId
+      meetingId: $meetingId
+      content: $content
+    ) {
+      ... on ErrorPayload {
+        error {
+          message
+        }
+      }
+      ...UpsertTeamPromptResponseMutation_meeting @relay(mask: false)
+    }
+  }
+`
+
+const UpsertTeamPromptResponseMutation: StandardMutation<TUpsertTeamPromptResponseMutation> = (
+  atmosphere,
+  variables,
+  {onError, onCompleted}
+) => {
+  return commitMutation<TUpsertTeamPromptResponseMutation>(atmosphere, {
+    mutation,
+    variables,
+    optimisticUpdater: (store) => {
+      const {viewerId} = atmosphere
+      const {meetingId, content} = variables
+      if (!meetingId) return
+      const meeting = store.get(meetingId)
+      if (!meeting) return
+      const phases = meeting.getLinkedRecords('phases')
+      if (!phases) return
+      const stages = phases[0]!.getLinkedRecords('stages')
+      if (!stages) return
+      const stage = stages.find(
+        (s) => s.getLinkedRecord('teamMember')?.getValue('userId') === viewerId
+      )
+      if (!stage) return
+      const response = stage.getLinkedRecord('response')
+      if (!response) {
+        // insertion
+        const newResponse = createProxyRecord(store, 'TeamPromptResponse', {
+          id: clientTempId(viewerId),
+          meetingId,
+          sortOrder: 0,
+          content
```

Comment:
-1 in the card component above it's referencing `plaintextContent`. 

either we need to add plaintextContent to this optimisticUpdater or in the component above we should remove `plaintextContent` and instead calculate that value.

Reply:
the query requires the field `plaintextContent`, but it's not provided here. which could cause an error if the component depends on it (e.g. add something like `plaintextContent.trim()` to the component & it'll complain that it's undefined.

when creating this new optimistic object, you'll need to do something like:
```ts
{
content: JSON.stringify(editorState.getJSON()),
plaintextContent: editor.state.doc.textContent // or however you get the plaintext
}
```

Reply:
oh got it. let's have this conversation in this PR - https://github.com/ParabolInc/parabol/pull/6504. I don't think we need an optimistic updater here at all.